syntax = "proto3";

package nsfw_detection_pb;

service NsfwDetectionProtoService {
  rpc DetectFromVideo(DetectFromVideoRequest) returns (VideoDetectionResponse);
  rpc DetectFromImage(DetectFromImageRequest) returns (ImageDetectionResponse);
};

enum Classifications {
  Safe = 0;
  ProbablySafe = 1;
  ProbablyUnsafe = 2;
  Unsafe = 3;
}

message UnsafeFrame {
  float second = 1;
  float nsfw_probability = 2;
}

message DetectFromVideoRequest {
  bytes video = 1;
  string video_format = 2;
};

message DetectFromImageRequest {
  bytes image = 1;
  string image_format = 2;
}

message VideoDetectionResponse {
  Classifications classification = 1;
  repeated UnsafeFrame unsafe_frames = 2;
}

message ImageDetectionResponse {
  Classifications classification = 1;
  float nsfw_probability = 2;
}
